addons/product/models/product_pricelist_item.py

```diff
@@ -554,13 +562,8 @@ def _compute_price_before_discount(self, *args, **kwargs):
 
         return pricelist_rule._compute_base_price(*args, **kwargs)
 
-    def _is_percentage(self):
+    def _show_discount(self):
         self and self.ensure_one()
-        return self.compute_price == 'percentage' or (
-            self.compute_price == 'formula'
-            and self.price_discount
-            and not self.price_surcharge
-            and not self.price_round
-            and not self.price_min_margin
-            and not self.price_max_margin
-        )
+        if not self.env.user.has_group('sale.group_discount_per_so_line'):
```

Comment:
Now that I think about it, not sure this group is also applied to public users, so it might be always considered as False for public/portal users (who might create SOlines through the ecommerce).

Reply:
I've created a task to handle this kind of issue globally since it arose multiple times recently, but for now we can stick to little 'hacks' like this one to check whether a feature group was enabled or not.